File: src/org/jgll/grammar/GrammarVisitAction.java
package org.jgll.grammar;

import org.jgll.grammar.slot.KeywordGrammarSlot;
import org.jgll.grammar.slot.LastGrammarSlot;
import org.jgll.grammar.slot.NonterminalGrammarSlot;

public interface GrammarVisitAction {

	public void visit(HeadGrammarSlot head);
	
	public void visit(NonterminalGrammarSlot slot);
	
	public void visit(TerminalGrammarSlot slot);
	
	public void visit(LastGrammarSlot slot);
	
	public void visit(KeywordGrammarSlot slot);
}

File: src-scala/org/jgll/grammar/GrammarVisitAction.scala
package org.jgll.grammar

import org.jgll.grammar.slot._

trait GrammarVisitActionTrait {
  self: HeadGrammarSlotTrait
   with NonterminalGrammarSlotTrait
   with TerminalGrammarSlotTrait
   with LastGrammarSlotTrait
   with KeywordGrammarSlotTrait =>
  trait GrammarVisitAction {

    def visit(head: HeadGrammarSlot): Unit

    def visit(slot: NonterminalGrammarSlot): Unit

    def visit(slot: TerminalGrammarSlot): Unit

    def visit(slot: LastGrammarSlot): Unit

    def visit(slot: KeywordGrammarSlot): Unit
  }
}